pkg/util/virtualmachine/virtualmachine.go

```diff
@@ -127,3 +132,45 @@ func SupportHotUnplugNic(vm *kubevirtv1.VirtualMachine) (bool, error) {
 
 	return false, fmt.Errorf("%s/%s doesn't support HotUnplugNic as none of its interfaces is hot-unpluggable: %s", vm.Namespace, vm.Name, strings.Join(errMsgs, ", "))
 }
+
+// CheckBlockRWXVolumeForVM checks whether the given PVC is a RWX volume (exclude Longhorn) and whether it is already used by other VMs.
+func CheckBlockRWXVolumeForVM(pvcCache v1.PersistentVolumeClaimCache, scCache ctlstoragev1.StorageClassCache, vmCache ctlkubevirtv1.VirtualMachineCache, pvcNS, pvcName, targetVMNS, targetVMName string) (bool, error) {
+	pvc, err := pvcCache.Get(pvcNS, pvcName)
+	if apierrors.IsNotFound(err) {
+		// means runtime creation, no need to check
+		return false, nil
+	}
+	if err != nil {
+		// any error here should be raised
+		return false, fmt.Errorf("failed to get PVC %s/%s, err: %s", pvcNS, pvcName, err)
+	}
+	targetAccessMode := pvc.Spec.AccessModes
+	targetProvisioner := util.GetProvisionedPVCProvisioner(pvc, scCache)
+	if volumeSupportRWXForVM(targetAccessMode, targetProvisioner) {
+		return true, nil
+	}
+	vms, err := vmCache.GetByIndex(indexeresutil.VMByPVCIndex, ref.Construct(pvcNS, pvcName))
+	if err != nil {
+		return false, fmt.Errorf("failed to get VMs by index: %s, PVC: %s/%s, err: %s", indexeresutil.VMByPVCIndex, pvcNS, pvcName, err)
+	}
+	for _, otherVM := range vms {
+		if otherVM.Namespace != targetVMNS || otherVM.Name != targetVMName {
+			return true, fmt.Errorf("the volume %s is already used by VM %s/%s", pvcName, otherVM.Namespace, otherVM.Name)
```

Comment:
should return here not be `false`? 
based on the logic if the PVC is not RWX or Longhorn as checked in `volumeSupportRWXForVM` then it means PVC cannot be attached to any more VM's anyways

Reply:
use new mechanism, no need